main/src/main/java/org/mobilitydata/gtfsvalidator/util/GeospatialUtil.java

```diff
@@ -66,98 +65,94 @@ private static DistanceCalculator getDistanceCalculator() {
+   * computed following the haversine formula. See
+   * https://locationtech.github.io/spatial4j/apidocs/org/locationtech/spatial4j/context/SpatialContext.html
+   * Note that points of origin (from) and destination (to) can be swapped. Result is expressed in
+   * meters.
    *
    * @param fromLat latitude of the first coordinates
    * @param fromLng longitude of the first coordinates
    * @param toLat latitude of the second coordinates
    * @param toLng longitude of the second coordinates
    * @return the calculation result in meters
    */
-  public static int distanceInMeterBetween(
+  public static double distanceInMeterBetween(
       double fromLat, double fromLng, double toLat, double toLng) {
     final ShapeFactory shapeFactory = getShapeFactory();
     final DistanceCalculator distanceCalculator = getDistanceCalculator();
     final Point origin = shapeFactory.pointXY(fromLng, fromLat);
     final Point destination = shapeFactory.pointXY(toLng, toLat);
-    return (int)
-        (DistanceUtils.DEG_TO_KM
-            * distanceCalculator.distance(origin, destination)
-            * KILOMETER_TO_METER_CONVERSION_FACTOR);
+    return DistanceUtils.DEG_TO_KM
+        * distanceCalculator.distance(origin, destination)
+        * KILOMETER_TO_METER_CONVERSION_FACTOR;
   }
 
   /**
-   * Returns a list of E052 errors for the given input, one for each stop that is too far from the
-   * trip shapePoints
+   * Returns a list of notices for the given input, one for each stop that is too far from the trip
+   * shapePoints
    *
-   * @param trip Trip for this GTFS trip
-   * @param stopTimes a map of StopTimes for a trip, sorted by stop_sequence
-   * @param shapePoints a map of ShapePoints for a trip, sorted by shape_pt_sequence
-   * @param stopTable a map of all stops (keyed on stop_id), needed to obtain the latitude and
-   *     longitude for each stop
+   * @param tripId trip_id for this GTFS trip
+   * @param stopTimes a list of StopTimes for a trip, sorted by stop_sequence
+   * @param shapeId the shape_id for this GTFS trip
+   * @param shapePoints a list of ShapePoints for a trip, sorted by shape_pt_sequence
+   * @param stopTable a container for all stops (keyed on stop_id), needed to obtain the latitude
+   *     and longitude for each stop on the trip
    * @param testedCache a cache for previously tested shape_id and stop_id pairs (keyed on
    *     shape_id+stop_id). If the combination of shape_id and stop_id appears in this set, we
    *     shouldn't test it again. Shapes and stops tested in this method execution will be added to
    *     this testedCache.
-   * @return a list of E052 errors, one for each stop that is too far from the trip shapePoints
+   * @return a list of notices, one for each stop that is too far from the trip shape
    */
   public static List<StopTooFarFromTripShapeNotice> checkStopsWithinTripShape(
```

Comment:
@lionel-nj As discussed let's refactor `checkStopsWithinTripShape()` into `StopTooFarFromTripShapeValidator`. It was originally put into the GeospatialUtil class in v1 in an attempt to adhere to the clean architecture principles (because it depends on an outside library). We aren't using clean architecture now, and given this returns a list of `StopTooFarFromTripShapeNotice` it can't be used in another validation rule, so it more naturally fits in the `StopTooFarFromTripShapeValidator` class. Also, this will allow us to more easily mock it to count the number of invocations to ensure we aren't accidentally processing trips more than once. We can't rely on the notice counts in unit tests for this rule to detect accidental executions because it uses a cache to avoid spamming the user with a large number of notices for the same stop and shape combination (e.g., if multiple trips re-use the same shape), and also as an optimization to avoid checking distance on these same combinations as well. But this means that if you process the same trip twice, you won't see more than the expected number of notices for processing the trip once - hence the need to mock the method to count invocations.